Patch to take CC, CXX, CFLAGS, LDFLAGS from the users (Cons) environment.
===================================================================
--- gtk/makefile	(revision 34521)
+++ gtk/makefile	(working copy)
@@ -7,13 +7,6 @@
 # Also works with ming32-make on Windows.
 
 .SUFFIXES: .cxx .c .o .h .a
-ifdef CLANG
-CC = clang
-CCOMP = clang
-else
-CC = g++
-CCOMP = gcc
-endif
 AR = ar
 RANLIB = touch
 
@@ -55,20 +48,20 @@
 endif
 
 ifdef DEBUG
-CXXFLAGS=-DDEBUG -g $(CXXBASEFLAGS) $(THREADFLAGS)
+CXXFLAGS += -DDEBUG -g $(CXXBASEFLAGS) $(THREADFLAGS)
+CFLAGS += -DDEBUG -g
 else
-CXXFLAGS=-DNDEBUG -Os $(CXXBASEFLAGS) $(THREADFLAGS)
+CXXFLAGS += -DNDEBUG -Os $(CXXBASEFLAGS) $(THREADFLAGS)
+CFLAGS += -DNDEBUG
 endif
 
-CFLAGS:=$(CXXFLAGS)
-
 CONFIGFLAGS:=$(shell pkg-config --cflags $(GTKVERSION))
 MARSHALLER=scintilla-marshal.o
 
 .cxx.o:
-	$(CC) $(CONFIGFLAGS) $(CXXFLAGS) -c $<
+	$(CXX) $(CONFIGFLAGS) $(CXXFLAGS) -c $<
 .c.o:
-	$(CCOMP) $(CONFIGFLAGS) $(CFLAGS) -w -c $<
+	$(CC) $(CONFIGFLAGS) $(CFLAGS) -w -c $<
 
 LEXOBJS:=$(addsuffix .o,$(basename $(notdir $(wildcard ../lexers/Lex*.cxx))))
 
